docs/content/docs/recipes/signals.md

```diff
@@ -0,0 +1,516 @@
+     */
+    private sendFocusSignal(hasFocus: boolean) {
+        this.signaler.submitSignal(
+            FocusTracker.focusSignalType,
+            { userId: this.audience.getMyself()?.userId, focus: hasFocus },
+        );
+    }
+
+    /*...*/
+}
+```
+`FocusTracker` differs from `MouseTracker` as it uses the `Signal Request` pattern. This is when a newly joining client requests a specific signal to be sent from other connected clients, so that the new client can receive pertinent information immediately after connecting to the container.
+
+To achieve this pattern, `FocusTracker` defines the `focusRequestType` that will be sent to request the focus status of all the connected clients:
+
+```typescript
+export class FocusTracker extends TypedEventEmitter<IFocusTrackerEvents> {
+    private static readonly focusRequestType = "focusRequest";
+
+    /*...*/
+}
+```
+`FocusTracker` then sends this signal request immediately after the new client connects to the container. Once again, the `onSignal` function is used to listen to this `focusRequestType` signal and each client responds to the signal request with their current focus status:
+
+```typescript
+export class FocusTracker extends TypedEventEmitter<IFocusTrackerEvents> {
+    private static readonly focusRequestType = "focusRequest";
+
+    /*...*/
+
+    public constructor(
+        container: IFluidContainer,
+        public readonly audience: IServiceAudience<IMember>,
+        private readonly signaler: Signaler,
+    ) {
+        super();
+
+        /*...*/
+
+        this.signaler.onSignal(FocusTracker.focusRequestType, () => {
+            this.sendFocusSignal(document.hasFocus());
+        });
+
+        container.on("connected", () => {
+            this.signaler.submitSignal(FocusTracker.focusRequestType);
+        });
+    }
+}
+```
+
+## View
+
+In `app.ts`, we create instances of the `MouseTracker` and the `FocusTracker` to use in our application:
+
+```typescript
+/*...*/
+
+async function start(): Promise<void> {
+    /*...*/
+
+    // Render presence information for audience members
+    const contentDiv = document.getElementById("focus-content") as HTMLDivElement;
+    const mouseContentDiv = document.getElementById("mouse-position") as HTMLDivElement;
+    const focusTracker = new FocusTracker(
+        container,
+        services.audience,
+        container.initialObjects.signaler as Signaler,
+    );
+    const mouseTracker = new MouseTracker(
+        services.audience,
+        container.initialObjects.signaler as Signaler,
+    );
+
+    /*...*/
+}
+
+start().catch(console.error);
+```
+
+The view then renders the focus data by using `renderFocusPresence`, which uses the local focus status map to display which users are in focus and which users are not. It also renders the mouse position data by using `renderMousePresence`, which uses the local position map to display the name of each user where they currently are in the window:
```

Comment:
Maybe show the snippets for the render functions, or a portion of them?  In particular would probably be good to show registering for the events to re-render.